### PR TITLE
Lifecycle phase not honouring retentions being set

### PR DESCRIPTION
**Pull request by johnsimons**
Fixes https://github.com/OctopusDeployLabs/terraform-provider-octopusdeploy/issues/489

### CODE REVIEW

**Review by scottmerchant — COMMENTED** [review]
This looks good to me - Obviously I'm not Terraform Provider (or Go) expert so maybe it'd be good to get someone else to look at it? Maybe @jbristowe or @slewis74?

File: octopusdeploy/schema_lifecycle_test.go [hunk 1]
```diff
@@ -18,10 +18,25 @@ func TestExpandLifecycle(t *testing.T) {
...
+			"unit":                "Days",
+		}}
+	tentacleRetention := []interface{}{
+		map[string]interface{}{
+			"quantity_to_keep":    2,
+			"should_keep_forever": false,
+			"unit":                "Items",
+		}}
+
 	resourceMap := map[string]interface{}{
```
Comment (johnsimons) [inline octopusdeploy/schema_lifecycle_test.go:34]: @jbristowe we probably should use `core.NewRetentionPeriod` here too?

File: octopusdeploy/schema_lifecycle_test.go [hunk 2]
```diff
@@ -18,10 +18,25 @@ func TestExpandLifecycle(t *testing.T) {
 	name := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
 	spaceID := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
 
+	releaseRetention := []interface{}{
+		map[string]interface{}{
+			"quantity_to_keep":    0,
+			"should_keep_forever": true,
+			"unit":                "Days",
+		}}
+	tentacleRetention := []interface{}{
```
Comment (johnsimons) [inline octopusdeploy/schema_lifecycle_test.go:27]: @jbristowe we probably should use core.NewRetentionPeriod here too?

File: octopusdeploy/schema_lifecycle_test.go [hunk 3]
```diff
@@ -18,10 +18,25 @@ func TestExpandLifecycle(t *testing.T) {
 	name := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
 	spaceID := acctest.RandStringFromCharSet(20, acctest.CharSetAlpha)
 
+	releaseRetention := []interface{}{
```
Comment (johnsimons) [inline octopusdeploy/schema_lifecycle_test.go:21]: @jbristowe we probably should use core.NewRetentionPeriod here too?

**Review by johnsimons — COMMENTED** [review]
LGTM

**Review by jbristowe — APPROVED** [review]

**Comment by scottmerchant** [comment]
[sc-41435]

**Pull request merged**